resources/[system]/chat/cl_chat.lua

```diff
@@ -19,16 +19,20 @@ AddEventHandler('chatMessage', function(author, color, text)
   if author ~= "" then
     table.insert(args, 1, author)
   end
-  SendNUIMessage({
-    type = 'ON_MESSAGE',
-    message = {
-      color = color,
-      multiline = true,
-      args = args
-    }
-  })
+  TriggerEvent("chat:cancelChatMessage")
+  if (not WasEventCanceled()) then
+    SendNUIMessage({
+      type = 'ON_MESSAGE',
+      message = {
+        color = color,
+        multiline = true,
+        args = args
+      }
+    })
+   end
 end)
 
+

```

Comment:
Note this random empty line was added by github. it wasn't there in my code 🤦‍♂️